### ISSUE TITLE
Under any parameter, loss is always reduced to nan during training

### ISSUE THREAD

**Issue by 177488ZL** [lucidrains/denoising-diffusion-pytorch#175]
Training under our dataset, img_ size=256 batch_ size=4 or img_ size=128 batch_ Size=16, the final result of all training is that loss is reduced to nan

**Comment by Harry-Stephanie** [comment]
same problem after 50000 training step

**Comment by snippler** [comment]
Any idea how to debug this issue?

**Comment by Echo-jyt** [comment]
> Training under our dataset, img_ size=256 batch_ size=4 or img_ size=128 batch_ Size=16, the final result of all training is that loss is reduced to nan

Hi, have you found the solution?

**Comment by 177488ZL** [comment]
no,I tried to train with various parameters for two weeks, and in the end, the loss was always null. My graphics card is 3080ti



At 2023-03-15 20:01:39, "Jin Yuntao" ***@***.***> wrote:

Training under our dataset, img_ size=256 batch_ size=4 or img_ size=128 batch_ Size=16, the final result of all training is that loss is reduced to nan

Hi, have you found the solution?

**Comment by Adrian744** [comment]
Did you double check your data ?(Make sure that all your data does not contain nan values)

**Comment by perkeje** [comment]
I had the same issue. If you are training with `amp = True`, be sure to run the script with `accelerate launch script.py`. That fixed my problem.

**Comment by mahakal001** [comment]
> I had the same issue. If you are training with `amp = True`, be sure to run the script with `accelerate launch script.py`. That fixed my problem.

Thank you for sharing.
